### PR TITLE
style: enable and baseline flakeheaven

### PR DESCRIPTION
**Pull request by aucampia**
# Summary of changes

This patch adds flakeheaven with a baseline, the baseline ignores all
existing errors and only reports on new errors.

The benefit with this approach is that we can get full flake8 on new
changes without having to first address the flake8 warnings with our
existing codebase.

Baselined errors can still be seen with:
- `flakeheaven lint --baseline /dev/null`
- `flake8`

This also adds a flakeheaven to GitHub Actions in a way which will
result in independent reporting.

Other changes:
- Simplified caching behaviour in GitHub Actions
- Removed flake8-black because it does not work well

  https://github.com/peterjc/flake8-black/issues/59
- Remove redundant/duplicated config for flake8

See also:
- https://github.com/RDFLib/rdflib/discussions/1927

# Checklist

- [x] Checked that there aren't other open pull requests for
  the same change.
- [x] Checked that all tests and type checking passes.
- [x] Considered granting [push permissions to the PR branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork),
  so maintainers can fix minor issues and keep your PR up to date.

### CODE REVIEW

**Review by gjhiggins — APPROVED** [review]

**Review by gjhiggins — APPROVED** [review]
still good :smile:

File: .github/workflows/validate.yaml [hunk 1]
```diff
@@ -81,6 +75,36 @@ jobs:
...
+        - task: "gha:flake8"
+          python-version: 3.8
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{env.DEFAULT_PYTHON}}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache pip
+        uses: actions/cache@v2
```
Comment (cclauss) [inline .github/workflows/validate.yaml:93]:
```suggestion
        uses: actions/cache@v3
```

File: .github/workflows/validate.yaml [hunk 2]
```diff
@@ -81,6 +75,36 @@ jobs:
...
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - task: "gha:flake8"
+          python-version: 3.8
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{env.DEFAULT_PYTHON}}
+        uses: actions/setup-python@v2
```
Comment (cclauss) [inline .github/workflows/validate.yaml:89]:
```suggestion
        uses: actions/setup-python@v3
```

File: .github/workflows/validate.yaml [hunk 3]
```diff
@@ -81,6 +75,36 @@ jobs:
...
+  extra-tasks:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - task: "gha:flake8"
+          python-version: 3.8
+    steps:
+      - uses: actions/checkout@v2
```
Comment (cclauss) [inline .github/workflows/validate.yaml:87]:
```suggestion
      - uses: actions/checkout@v3
```

File: requirements.dev.txt [hunk 4]
```diff
@@ -11,3 +9,4 @@ pytest-cov
 pytest-subtests
 types-setuptools
 -r docs/sphinx-requirements.txt
+-r requirements.flake8.txt
```
Comment (cclauss) [inline requirements.dev.txt:12]:
```suggestion
-r requirements-flake8.txt
```

**Review by cclauss — APPROVED** [review]

File: requirements.dev.txt (hunk 4 above)
Comment (aucampia) [reply requirements.dev.txt:12]: While I agree that `requirements-???.txt` is more conventional, I will leave this for now so it is in line with the rest of the files, I will maybe make a separate PR to change them all when I can.

**Comment by aucampia** [comment]
Thanks for the review and suggestions @cclauss.

**Pull request merged**